generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -106,7 +108,10 @@ open class FunctionDataFetcher(
      */
     protected open fun mapParameterToValue(param: KParameter, environment: DataFetchingEnvironment): Pair<KParameter, Any?>? =
         when {
-            param.isGraphQLContext() -> param to environment.getContext()
+            param.isGraphQLContext() -> {
+                logger.warn("GraphQLContext interface injection is deprecated. Please use DataFetchingEnvironment to retrieve ${param.getName()}.")
```

Comment:
Depending on how often this feature is used this could end up spamming the logs A LOT. I'm wondering whether we should log the warnings during the schema generation process instead.

Reply:
Agree and i was actually just thinking the same thing.  I am unsure where that code lives but will do some hunting.  If you had a place in mind please let me know.

Reply:
I refactored the PR so that the warning is added during schema generation.  Let me know what you think @dariuszkuc